From eb22af86265a027c5aa64086f5978123fe6a294d Mon Sep 17 00:00:00 2001
From: Han Gao <gaohan@iscas.ac.cn>
Date: Sun, 10 May 2026 23:42:39 +0800
Subject: [PATCH 92/99] RUYI: riscv: dts: pine64-starpro64: enable EMMC and SD
 card

Enable the on-board eMMC and SD card slot on the Pine64 StarPro64 board.

The eMMC cannot run reliably in HS400 mode on this board, so override
the SoC default to use HS200 1.8V instead.

The SD card detect signal (SD0_DET net) is wired to SoC ball F14
(mode_set0 pin); when this pin is muxed to GPIO function it appears as
gpioA[13]. Add a pinctrl group selecting GPIO function with input
enable and pull-up bias for the mode_set0 pin, and wire cd-gpios to
gpioA[13] so the SDHCI driver uses GPIO-based card detection.

Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 .../dts/eswin/eic7700-pine64-starpro64.dts    | 29 +++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
index 00a76b47cb77..a17e0e7d8917 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
+++ b/arch/riscv/boot/dts/eswin/eic7700-pine64-starpro64.dts
@@ -12,6 +12,8 @@ / {
 	model = "Pine64 StarPro64";
 
 	aliases {
+		mmc0 = &sdhci_emmc;
+		mmc1 = &sdio0;
 		serial0 = &uart0;
 		ethernet0 = &gmac0;
 		ethernet1 = &gmac1;
@@ -40,6 +42,15 @@ &pinctrl {
 	status = "okay";
 	vrgmii-supply = <&vcc_1v8>;
 
+	pinctrl_gpio13: gpio13-grp {
+		gpio13-pins {
+			pins = "mode_set0";
+			function = "gpio";
+			input-enable;
+			bias-pull-up;
+		};
+	};
+
 	pinctrl_gpio16: gpio16-grp {
 		gpio16-pins {
 			pins = "mode_set3";
@@ -102,6 +113,24 @@ gmac1_phy0: ethernet-phy@0 {
 	};
 };
 
+&sdhci_emmc {
+	status = "okay";
+	/*
+	 * The eMMC data strobe pin is wired to RST on this board, so HS400
+	 * (which requires a working data strobe) cannot work. Cap the eMMC
+	 * at HS200 1.8V.
+	 */
+	/delete-property/ mmc-hs400-1_8v;
+	mmc-hs200-1_8v;
+};
+
+&sdio0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio13>;
+	cd-gpios = <&gpioA 13 GPIO_ACTIVE_LOW>;
+};
+
 &uart0 {
 	status = "okay";
 };
-- 
2.53.0

